services/management/ia/templates_service.go

```diff
@@ -61,25 +69,41 @@ type TemplatesService struct {
 	l                 *logrus.Entry
 	userTemplatesPath string
 
+	host       string
+	publicKeys []string
+
 	rw        sync.RWMutex
 	templates map[string]templateInfo
 }
 
 // NewTemplatesService creates a new TemplatesService.
-func NewTemplatesService(db *reform.DB) *TemplatesService {
+func NewTemplatesService(db *reform.DB) (*TemplatesService, error) {
 	l := logrus.WithField("component", "management/ia/templates")
 
 	err := dir.CreateDataDir(templatesDir, "pmm", "pmm", dirPerm)
 	if err != nil {
 		l.Error(err)
 	}
 
-	return &TemplatesService{
+	host, err := envvars.GetSAASHost()
+	if err != nil {
+		return nil, err
+	}
+
+	s := &TemplatesService{
 		db:                db,
 		l:                 l,
 		userTemplatesPath: templatesDir,
+		host:              host,
 		templates:         make(map[string]templateInfo),
 	}
+
+	if k := os.Getenv(envPublicKey); k != "" {
+		s.publicKeys = strings.Split(k, ",")
+		l.Warnf("Public keys changed to %q.", k)
+	}
```

Comment:
IMHO this part also can be moved somewhere. I don't remember if I told earlier, but maybe we can create some signature service. 

Anyway, it's just thoughts, let's keep it as is for now.

Reply:
added a todo comment for it